doc/examples/example_mac.md

```diff
@@ -0,0 +1,60 @@
+# Message Authentication Code (MAC)
+
+# Message Authentication Code Interface
+
+All the MAC classes (currently only one) implement the methods listed below.
+Algorithm specific paramaters are specified via the constructor.
+
+```cpp
+class MessageAuthenticationCode {
+public:
+    ...
+
+    virtual void update(const std::vector<uint8_t>& message) = 0;
+
+    virtual std::vector<uint8_t> finish() = 0;
+
+    virtual void verify(const std::vector<uint8_t> &macValue) = 0;
+};
+```
+
+# HMAC
+
+## HMAC Creation
+
+The following example shows how to create an HMAC.
+
+```cpp
+std::vector<uint8_t> key = {'1', '2', '3', '4', '5', '6'};
+std::vector<uint8_t> msg = {'m', 'e', 's', 's', 'a', 'g', 'e'};
+
+mococrw::HMAC hmac = mococrw::HMAC(openssl::DigestTypes::SHA256, key);
+hmac.update(msg);
+std::vector<uint8_t> mac = hmac.finish();
+```
+
+## HMAC Verification
+
+The following example shows how an HMAC can be verified.
```

Comment:
Please add a note that this is a constant-time comparison, and users should always use .verify() and not attempt to compute the mac again and compare it using a string or byte comparison.